src/reason-parser/reason_pprint_ast.ml

```diff
@@ -7576,29 +7576,40 @@ let printer = object(self:'self)
     | Pmod_constraint _
     | Pmod_structure _ -> self#simple_module_expr x
 
+  (* We print differently top level extensions than other extensions
+    in structure_items or expressions. In this case we print it with %%name *)
+  method top_level_structure_item item =
+    match item.pstr_desc with
+      | Pstr_extension ((extension, PStr [item]), attrs) ->
+          begin match item.pstr_desc with
+            (* In case of a value, the extension gets inlined `let%private a = 1` *)
+            | Pstr_value (rf, vb_list) -> self#bindings ~extension (rf, vb_list)
+            | _ -> self#attach_std_item_attrs attrs (self#payload "%%" extension (PStr [item]))
+          end
+      | _ -> self#structure_item item
 
   method structure structureItems =
     match structureItems with
     | [] -> atom ""
-    | first::_ as structureItems ->
+    | first :: _ as structureItems ->
       let last = match (List.rev structureItems) with | last::_ -> last | [] -> assert false in
       let loc_start = first.pstr_loc.loc_start in
       let loc_end = last.pstr_loc.loc_end in
       let items =
         groupAndPrint
-          ~xf:self#structure_item
+          ~xf:self#top_level_structure_item
```

Comment:
This looks fishy.

Reply:
Probably a bad name, but the logic is to check for a specific combination of AST such as: `structure -> structure_item -> Pexp_extension`